File: backend/internal/constants/node.go
package constants

type NodeType string

const (
	NodeTypeIntake   NodeType = "intake"
	NodeTypeExhaust  NodeType = "exhaust"
	NodeTypeWorkface NodeType = "workface"
	NodeTypeJunction NodeType = "junction"
)

type NodeStatus string

const (
	NodeStatusActive   NodeStatus = "active"
	NodeStatusInactive NodeStatus = "inactive"
	NodeStatusBlocked  NodeStatus = "blocked"
)

func ValidNodeType(value string) bool {
	switch NodeType(value) {
	case NodeTypeIntake, NodeTypeExhaust, NodeTypeWorkface, NodeTypeJunction:
		return true
	default:
		return false
	}
}

func ValidNodeStatus(value string) bool {
	switch NodeStatus(value) {
	case NodeStatusActive, NodeStatusInactive, NodeStatusBlocked:
		return true
	default:
		return false
	}
}

File: frontend/src/types/node.ts
export type NodeType = 'intake' | 'exhaust' | 'workface' | 'junction';
export type NodeStatus = 'active' | 'inactive' | 'blocked';

export interface VentilationNode {
  id: number;
  code: string;
  node_type: NodeType;
  elevation_m: number;
  required_airflow_m3s: number;
  pressure_pa: number;
  status: NodeStatus;
  created_at: string;
  updated_at: string;
}

export type CreateNodeInput = Omit<VentilationNode, 'id' | 'created_at' | 'updated_at'>;
export type UpdateNodeInput = Omit<CreateNodeInput, 'code'>;

export interface NetworkIssue {
  code: string;
  entity_type: string;
  entity_id: number;
  message: string;
  severity: 'warning' | 'critical';
}

export interface NetworkValidation {
  valid: boolean;
  node_count: number;
  edge_count: number;
  intake_count: number;
  exhaust_count: number;
  issues: NetworkIssue[];
}
